File: src/main/java/me/jameschan/burrow/kernel/command/builtin/RootCommand.java
package me.jameschan.burrow.kernel.command.builtin;

import me.jameschan.burrow.kernel.command.Command;
import me.jameschan.burrow.kernel.common.ExitCode;
import me.jameschan.burrow.kernel.context.ChamberContext;
import me.jameschan.burrow.kernel.context.RequestContext;
import me.jameschan.burrow.kernel.furniture.annotation.CommandType;
import picocli.CommandLine;

@CommandLine.Command(name = "root", description = "Print the root directory of the application.")
@CommandType(CommandType.BUILTIN)
public class RootCommand extends Command {
  public RootCommand(final RequestContext requestContext) {
    super(requestContext);
  }

  @Override
  public Integer call() {
    buffer.append(getRootDirectoryAbsolutePath(context));
    return ExitCode.SUCCESS;
  }

  public static String getRootDirectoryAbsolutePath(final ChamberContext context) {
    return context.getRootDir().toString();
  }
}

File: src/main/kotlin/carton/standard/command/RootCommand.kt
package burrow.carton.standard.command

import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine
import picocli.CommandLine.ExitCode

@CommandLine.Command(
    name = "root",
    description = [
        "Displays the absolute path to the root directory of the current " +
                "chamber."
    ]
)
class RootCommand(data: CommandData) : Command(data) {
    override fun call(): Int {
        stdout.println(chamber.rootPath)

        return ExitCode.OK
    }
}